opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -614,34 +615,42 @@ namespace Opm {
             std::size_t completion_index = 0;
             well_perf_data_[well_index].clear();
             well_perf_data_[well_index].reserve(well.getConnections().size());
+            CheckDistributedWellConnections checker(well, *local_parallel_well_info_[well_index]);
+            bool hasFirstPerforation = false;
+            bool firstOpenCompletion = true;
+
             for (const auto& completion : well.getConnections()) {
+                const int i = completion.getI();
+                const int j = completion.getJ();
+                const int k = completion.getK();
+                const int cart_grid_indx = i + cartDims[0] * (j + cartDims[1] * k);
+                const int active_index = cartesian_to_compressed_[cart_grid_indx];
```

Comment:
If you're touching this anyway, now would be a good time to use [`Connection::global_index()`](https://github.com/OPM/opm-common/blob/1ef1646fb191b71a62dbef184942de9d93a21c8c/opm/parser/eclipse/EclipseState/Schedule/Well/Connection.hpp#L108) to save on some arithmetic here:
```C++
const auto active_index =
    this->cartesian_to_compressed_[completion.global_index()];
```